test/e2e/rollout_test.go

```diff
@@ -472,6 +473,162 @@ var _ = Describe("placing wrapped resources using a CRP", Ordered, func() {
+				if work.Status.ManifestConditions != nil {
+					work.Status.ManifestConditions = nil
+				} else {
+					meta.SetStatusCondition(&work.Status.Conditions, metav1.Condition{
+						Type:   placementv1beta1.WorkConditionTypeAvailable,
+						Status: metav1.ConditionFalse,
+						Reason: "WorkNotAvailable",
+					})
+				}
+				Expect(hubClient.Status().Update(ctx, &work)).Should(Succeed(), "Failed to update the work")
+			}
+		})
+
+		It("change the image name in deployment, to roll over the resourcesnapshot", func() {
+			crsList := &placementv1beta1.ClusterResourceSnapshotList{}
+			Expect(hubClient.List(ctx, crsList, client.MatchingLabels{placementv1beta1.CRPTrackingLabel: crpName})).Should(Succeed(), "Failed to list the resourcesnapshot")
+			Expect(len(crsList.Items) == 1).Should(BeTrue())
+			oldCRS := crsList.Items[0].Name
+			Expect(hubClient.Get(ctx, types.NamespacedName{Name: testDeployment.Name, Namespace: testDeployment.Namespace}, &testDeployment)).Should(Succeed(), "Failed to get deployment")
+			testDeployment.Spec.Template.Spec.Containers[0].Image = "extra-snapshot"
+			Expect(hubClient.Update(ctx, &testDeployment)).Should(Succeed(), "Failed to change the image name in deployment")
+			// wait for the new resourcesnapshot to be created
+			Eventually(func() bool {
+				Expect(hubClient.List(ctx, crsList, client.MatchingLabels{placementv1beta1.CRPTrackingLabel: crpName})).Should(Succeed(), "Failed to list the resourcesnapshot")
+				Expect(len(crsList.Items) == 1).Should(BeTrue())
+				return crsList.Items[0].Name != oldCRS
+			}, eventuallyDuration, eventuallyInterval).Should(BeTrue(), "Failed to remove the old resourcensnapshot")
+		})
+
+		It("update work to trigger a work generator reconcile", func() {
+			for idx := range allMemberClusters {
+				memberCluster := allMemberClusters[idx].ClusterName
+				namespaceName := fmt.Sprintf(utils.NamespaceNameFormat, memberCluster)
+				workName := fmt.Sprintf(placementv1beta1.FirstWorkNameFmt, crpName)
+				work := placementv1beta1.Work{}
+				Expect(hubClient.Get(ctx, types.NamespacedName{Name: workName, Namespace: namespaceName}, &work)).Should(Succeed(), "Failed to get the work")
+				if work.Status.ManifestConditions != nil {
+					work.Status.ManifestConditions = nil
+				} else {
+					meta.SetStatusCondition(&work.Status.Conditions, metav1.Condition{
+						Type:   placementv1beta1.WorkConditionTypeAvailable,
+						Status: metav1.ConditionFalse,
+						Reason: "WorkNotAvailable",
+					})
+				}
+				Expect(hubClient.Status().Update(ctx, &work)).Should(Succeed(), "Failed to update the work")
+			}
+		})
+
+		It("change the image name in deployment, to make it available again", func() {
+			Eventually(func() error {
+				err := hubClient.Get(ctx, types.NamespacedName{Name: testDeployment.Name, Namespace: testDeployment.Namespace}, &testDeployment)
+				if err != nil {
+					return err
+				}
+				testDeployment.Spec.Template.Spec.Containers[0].Image = "1.26.2"
```

Comment:
Ah, Ryan, I think this would be still be an invalid image name? But it does align with the situation Infosys encountered (which is, even if the old snapshot has been removed and all clusters are failing, updates should still be processed).